### PR TITLE
Test Application Creation via Git source components

### PR DESCRIPTION
**Pull request by dheerajodha**
## Fixes 

Fixes: https://issues.redhat.com/browse/HAC-1291

## Description

Adds a UI test that automates the creation and deletion of the component of an application by using a Git URL as a source.

## Type of change

- [x] Other (please describe):
Adds UI test using cypress.

## How to test or reproduce?
Please read the readme in the integration tests folder.

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { createApplicationPagePO } from '../pageObjects/createApplication-po';
+import { buildApplicationWithGitSourcePagePO } from '../pageObjects/createApplication-po';
+import { AbstractWizardPage } from './AbstractWizardPage';
+export class BuildApplicationWithGitSourcePage extends AbstractWizardPage {
```
Comment (Katka92) [inline integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts:4]: What about merging this class with [https://github.com/openshift/hac-dev/blob/main/integration-tests/support/pages/AddComponentPage.ts](https://github.com/openshift/hac-dev/blob/main/integration-tests/support/pages/AddComponentPage.ts ) as it already contains some logic from the same page?

File: integration-tests/support/pageObjects/createApplication-po.ts [hunk 2]
```diff
@@ -31,3 +31,11 @@ export const componentsListPagePO = {
   applicationName: 'pf-c-title pf-m-4xl hacDev-page__heading',
   items: '[pf-c-data-list__item]',
 };
+
+export const buildApplicationWithGitSourcePagePO = {
```
Comment (Katka92) [inline integration-tests/support/pageObjects/createApplication-po.ts:35]: Maybe worth merging with *addComponentPagePO* as it already contains some elements of this page.

File: integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  clickGitOptions() {
+    cy.contains(buildApplicationWithGitSourcePagePO.gitOptions).click();
+  }
+
+  setGitReference(gitReference: string) {
+    this.clearGitReference();
+    cy.get(buildApplicationWithGitSourcePagePO.gitReference).type(gitReference);
+  }
+
+  clearGitReference() {
```
Comment (Katka92) [inline integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts:27]: I'm not sure if we need this logic to be separated into an extra method as it contains the reference usable only with this particular element. Applies to other "clear" methods too.

**Review by Katka92 — CHANGES_REQUESTED** [review]

File: integration-tests/tests/create-application-from-git-source.spec.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// <reference types="cypress" />
+import { AddComponentPage } from '../support/pages/AddComponentPage';
+import { ApplicationComponentPage } from '../support/pages/ApplicationComponentsPage';
+import { ComponentSamplesPage } from '../support/pages/ComponentSamplesPage';
+import { CreateApplicationPage } from '../support/pages/CreateApplicationPage';
+import { BuildApplicationWithGitSourcePage } from '../support/pages/BuildApplicationWithGitSourcePage';
+import { Common } from '../utils/Common';
+
+describe('Create Application from Git Source', () => {
```
Comment (jrichter1) [inline integration-tests/tests/create-application-from-git-source.spec.ts:9]: it's the component that is being created from git, not the application, so the terminology is a bit off

File: integration-tests/tests/create-application-from-git-source.spec.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  const applicationName = `test-app${new Date().getTime() / 1000}`;
+
+  it('Creating a Quarkus component', () => {
+    //set application name
+    const createApplicationPage = new CreateApplicationPage();
+    createApplicationPage.clickCreateApplication();
+    createApplicationPage.setApplicationName(applicationName);
+    createApplicationPage.clickNext();
```
Comment (jrichter1) [inline integration-tests/tests/create-application-from-git-source.spec.ts:17]: creating the application would make more sense in the `before` hook

File: integration-tests/tests/create-application-from-git-source.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  it('Delete java quarkus component', () => {
+    //Open components page
+    Common.openApplicationURL(applicationName);
+    //Review component page
+    const applicationPage = new ApplicationComponentPage();
+    applicationPage.deleteComponent('1-java-quarkus');
+
+    //Check if application does not exists
+    applicationPage.createdApplicationNotExists('1-java-quarkus');
+  });
```
Comment (jrichter1) [inline integration-tests/tests/create-application-from-git-source.spec.ts:43]: we already have this exact test, so this looks more like a cleanup step, which should be in the `after` hook

File: integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts (hunk 1 above)
Comment (dheerajodha) [reply integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts]: Sure, I've deleted this file and moved the methods to the AddComponentPage class. Can you please take a look again and see if it looks good?

File: integration-tests/tests/create-application-from-git-source.spec.ts (hunk 4 above)
Comment (dheerajodha) [reply integration-tests/tests/create-application-from-git-source.spec.ts]: Makes sense :+1: thanks

File: integration-tests/tests/create-application-from-git-source.spec.ts (hunk 6 above)
Comment (dheerajodha) [reply integration-tests/tests/create-application-from-git-source.spec.ts]: done, thanks!

File: integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts (hunk 1 above)
Comment (Katka92) [reply integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts]: Nice, thanks :+1:

File: integration-tests/support/pageObjects/createApplication-po.ts (hunk 2 above)
Comment (Katka92) [reply integration-tests/support/pageObjects/createApplication-po.ts]: Thanks!

File: integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts (hunk 3 above)
Comment (Katka92) [reply integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts]: Looks better to me, thanks. :slightly_smiling_face:

File: integration-tests/support/pageObjects/createApplication-po.ts [hunk 7]
```diff
@@ -1,6 +1,12 @@
 export const addComponentPagePO = {
   samples: 'Start with a sample.',
   addComponent: '[data-testid="add-component"]',
+  enterSource: '[id="form-input-source-field"]',
+  validated: '[id="form-input-source-field-helper"]',
+  gitOptions: 'Git options',
+  gitReference: '[id="form-input-git-reference-field"]',
+  contextDir: '[id="form-input-git-contextDir-field"]',
+  next: '[data-test="submit-button"]'
```
Comment (rohitkrai03) [inline integration-tests/support/pageObjects/createApplication-po.ts:9]: Shouldn't we be consistent here. Some have `data-testid`, some have `data-test` and some `id`?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: integration-tests/support/pageObjects/createApplication-po.ts (hunk 7 above)
Comment (dheerajodha) [reply integration-tests/support/pageObjects/createApplication-po.ts:8]: Sure, I've made some changes in the corresponding files of the components by adding a new `data-testid` field as it wasn't present before, and then updated the test accordingly.

Comment (rohitkrai03) [reply integration-tests/support/pageObjects/createApplication-po.ts:8]: @jrichter1 Do you remember what is the standard (`data-test` or `data-testid`) we use across console?

Comment (jrichter1) [reply integration-tests/support/pageObjects/createApplication-po.ts:8]: there is the legacy `data-test-id` and the current  is `data-test`

Comment (dheerajodha) [reply integration-tests/support/pageObjects/createApplication-po.ts:8]: so should I replace `data-testid` with `data-test`?

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**